safe/gui/tools/wizard/step_kw33_multi_classifications.py

```diff
@@ -0,0 +1,534 @@
+        """Setup the UI for QTextEdit to show the current state."""
+        layer_mode = self.parent.step_kw_layermode.selected_layermode()
+        right_panel_heading = QLabel(tr('Status'))
+        right_panel_heading.setFont(big_font)
+        self.right_layout.addWidget(right_panel_heading)
+
+        message = m.Message()
+        if layer_mode == layer_mode_continuous:
+            unit = self.parent.step_kw_unit.selected_unit()
+            title = tr('Thresholds')
+        else:
+            unit = None
+            title = tr('Value maps')
+
+        message.add(m.Heading(title, **INFO_STYLE))
+
+        for i in range(len(self.exposures)):
+            message.add(m.Text(self.exposure_labels[i]))
+            table = m.Table(style_class='table table-condensed table-striped')
+            header = m.Row()
+            header.add(m.Cell(tr('Class name')))
+            header.add(m.Cell(tr('Minimum')))
+            header.add(m.Cell(tr('Maximum')))
+            table.add(header)
+            classification = self.get_classification(
+                self.exposure_combo_boxes[i])
+            default_thresholds = classification_thresholds(classification, unit)
+            thresholds = self.thresholds.get(self.exposures[i]['key'])
+            if not thresholds:
+                thresholds = default_thresholds
+            classes = classification.get('classes')
+            # Sort by value, put the lowest first
+            classes = sorted(classes, key=lambda k: k['value'])
+            for the_class in classes:
+                threshold = thresholds[the_class['key']]
+                row = m.Row()
+                row.add(m.Cell(the_class['name']))
+                row.add(m.Cell(threshold[0]))
+                row.add(m.Cell(threshold[1]))
+                table.add(row)
+            #
+            # for class_key, threshold in thresholds.items():
+            #     row = m.Row()
+            #     row.add(m.Cell(class_key))
+            #     row.add(m.Cell(threshold[0]))
+            #     row.add(m.Cell(threshold[1]))
+            #     table.add(row)
+            message.add(table)
+
+        status_text_edit = QTextBrowser(None)
+        # status_text_edit = MessageViewer(None)
+        html_string = html_header() + message.to_html() + html_footer()
+        status_text_edit.setHtml(html_string)
+        self.right_layout.addWidget(status_text_edit)
+
+    def set_widgets(self):
+        """Set widgets on the Multi classification step."""
+        # Set the step description
+        self.set_wizard_step_description()
+
+        # Set the left panel
+        self.setup_left_panel()
+
+        # Set the right panel, for the beginning show the viewer
+        self.show_current_state()
+
+    def clear(self):
+        """Clear current state."""
+        clear_layout(self.left_layout)
+        clear_layout(self.right_layout)
+
+    def get_current_state(self):
+        """Obtain current classification and value map / threshold."""
```

Comment:
See https://github.com/inasafe/inasafe/pull/3703/files#diff-bb8df5d8797dfba301be3a89226ca551R412

I even put inner function inside the method.